libcudacxx/include/cuda/__mdspan/host_device_accessor.h

```diff
@@ -69,25 +82,28 @@ inline constexpr bool is_host_device_managed_accessor_v =
  **********************************************************************************************************************/
 
 template <typename _Accessor>
```

Comment:
`_Accessor` needs to be constrained here so that `data_handle_type` is `element_type*`.  Otherwise, the run-time "this pointer points to a valid allocation in the given space" check won't compile.  I'll express the constraint in C++ >= 20 form just for brevity and clarity.

```suggestion
template <typename _Accessor>
  requires(std::is_same_v<typename _Accessor::data_handle_type, typename _Accessor::element_type*>)
```

Reply:
I'm not 100% sure that host/device accessors are limited to pointers. I protected the check with `if constexpr (is_pointer<data_handle_type>`

Reply:
@fbusato Ah, sorry, that's right, the check is already protected.  Never mind : - ) .